spec/lib/services/pollers-api-service-spec.js

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+"use strict";
+
+describe("Http.Services.Api.Pollers", function () {
+    var pollerService;
+    var waterline;
+    var taskProtocol;
+
+    before("Http.Services.Api.Pollers before", function() {
+        helper.setupInjector([
+            helper.require("/lib/services/pollers-api-service.js")
+        ]);
+
+        waterline = helper.injector.get('Services.Waterline');
+        waterline.workitems = {
+            needByIdentifier: sinon.stub().resolves(),
+            find: sinon.stub().resolves([]),
+            create: sinon.stub().resolves(),
+            updateByIdentifier: sinon.stub().resolves(),
+            destroyByIdentifier: sinon.stub().resolves()
+        };
+
+	    taskProtocol = helper.injector.get("Protocol.Task");
+        taskProtocol.requestPollerCache = sinon.stub();
+
+        pollerService = helper.injector.get("Http.Services.Api.Pollers");
+
+    });
+
+    describe("getPollers", function() {
+        it('should expose the appropriate methods', function() {
+            pollerService.should.have.property('getPollers')
+                .that.is.a('function').with.length(1);
+        });
+
+        it('Run getPollers', function() {
+            var mockPoller = [{
+                id: '4532',
+                name: 'Pollers.IPMI',
+                config: {}
+            }];
+
+            waterline.workitems.find.resolves([mockPoller]);
+            return pollerService.getPollers().then(function (pollers) {
+                expect(pollers).to.deep.equal([mockPoller]);
+            });
+        });
+       
+        it('should return error message  if no poller informations are found', function () {
+            var mockPoller = {"message":"NotFoundError: Could not find workitem with identifier"};
+            waterline.workitems.find.resolves(mockPoller);
+            return pollerService.getPollers().should.eventually.become(mockPoller);
+        });
+    });
+
+    describe("getPollersById", function() {
+        it('should expose the appropriate methods', function() {
+            pollerService.should.have.property('getPollersById')
+            .that.is.a('function').with.length(1);
+        });
+
+        it('Run getPollersById', function() {
+            var mockPoller = [{
+                id: '4532',
+                name: 'Pollers.IPMI',
+                config: {}
+            }];
+            waterline.workitems.needByIdentifier.resolves([mockPoller]);
+            return pollerService.getPollersById().then(function (pollers) {
```

Comment:
Since you are testing getPollersById, so I think you should mock at least 2 pollers in db and then pass a poller id into getPollersById and expect to get the correct poller, something like this:
```javascript
var mockPoller = [{
    id: '4532',
    name: 'Pollers.IPMI',
    config: {}
}, {
    id: '1234',
    name: 'Pollers.TEST',
    config: {}
}];
waterline.workitems.needByIdentifier.withArgs({'id': '4532'}).resolves(mockPoller[0]);
waterline.workitems.needByIdentifier.withArgs({'id': '1234'}).resolves(mockPoller[1]);
return Promise.all(['4532', '1234'].map(function(id, index) {
    return pollerService.getPollersById(index).then(function (poller) {
    expect(poller).to.deep.equal(mockPoller[index]);
});
```